### PR TITLE
hack/update-codegen.sh: add -mod=vendor to work with recent Golang

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -23,32 +23,42 @@ API_PKG="${THIS_REPO}/pkg/apis/migration/v1alpha1"
...
 
-go run ./vendor/sigs.k8s.io/controller-tools/cmd/controller-gen \
+mkdir -p _output
+go run -mod=vendor ./vendor/sigs.k8s.io/controller-tools/cmd/controller-gen \
   schemapatch:manifests="${THIS_REPO_ABSOLUTE}/manifests" \
   paths="${THIS_REPO_ABSOLUTE}/pkg/apis/migration/v1alpha1" \
   output:dir="${THIS_REPO_ABSOLUTE}/manifests"
 
-go run ./vendor/k8s.io/code-generator/cmd/client-gen \
+curl -s -f -L https://github.com/krishicks/yaml-patch/releases/download/v0.0.10/yaml_patch_$(go env GOHOSTOS) -o _output/yaml-patch
```
Comment (deads2k) [inline hack/update-codegen.sh:32]: comment about what this is doing.

File: manifests/storage_state_crd.yaml-patch [hunk 2]
```diff
@@ -0,0 +1,6 @@
+- op: add
+  path: /spec/versions/name=v1alpha1/schema/openAPIV3Schema/properties/metadata/properties
```
Comment (deads2k) [inline manifests/storage_state_crd.yaml-patch:2]: what does this do?

Comment (sttts) [reply manifests/storage_state_crd.yaml-patch:2]: readd the metadata.name schema. It looks like it was just manually added to the CRD manifest and controller-tools will override that.

File: hack/update-codegen.sh (hunk 1 above)
Comment (sttts) [reply hack/update-codegen.sh:34]: done

File: pkg/apis/migration/v1alpha1/types.go [hunk 3]
```diff
@@ -120,7 +120,6 @@ type StorageVersionMigrationList struct {
 // The state of the storage of a specific resource.
 type StorageState struct {
 	metav1.TypeMeta `json:",inline"`
-	// The name must be "<.spec.resource.resouce>.<.spec.resource.group>".
```
Comment (sttts) [inline pkg/apis/migration/v1alpha1/types.go:123]: @deads2k this was wrong. You cannot add a custom comment to metadata.

**Comment by deads2k** [comment]
the more you know...

/lgtm
/approve

**Pull request merged**